Source/IntegrationTests/TestFiles/LitTests/LitTest/dafny2/SmallestMissingNumber-functional.dfy

```diff
@@ -90,7 +90,7 @@ function SMN''(xs: List<nat>, n: nat, len: nat): nat
       SMN''(R, n + llen, len - llen)
 }
 
-function Split(xs: List<nat>, b: nat): (List<nat>, List<nat>)
+opaque function Split(xs: List<nat>, b: nat): (List<nat>, List<nat>)
```

Comment:
I often make the decision based on if the property is one I think all users of the function are going want. If it is, then I add it as a postcondition of the function. Otherwise, I make it into a separate lemma.

In this case, the property stated by `Split_Correct` is relevant only in situations where the input has no duplicates. Since the `Split` function doesn't require the no-duplicates property, I think the lemma is best stated separately.